### PR TITLE
Some follow-up on #617.

### PR DESCRIPTION
**Pull request by jameshaydon**

### CODE REVIEW

File: src/Radicle/Internal/ConcurrentMap.hs [hunk 1]
```diff
@@ -26,28 +26,26 @@ empty = CMap <$> newMVar Map.empty
 
 -- | Atomically lookup a value.
 lookup :: Ord k => k -> CMap k v -> IO (Maybe v)
-lookup k (CMap m_) = withMVar m_ $ \m -> do
-   case Map.lookup k m of
-     Nothing -> pure Nothing
-     Just v_ -> readMVar v_
+lookup k m = modifyValue k m (\x -> pure (x,x))
```
Comment (MeBrei) [inline src/Radicle/Internal/ConcurrentMap.hs]: If `Map.lookup k m` returns `Nothing` wont this now lead to it being inserted in `modifyValue`?

Comment (jameshaydon) [reply src/Radicle/Internal/ConcurrentMap.hs]: I'm not sure what you mean. Are you commenting on the old code or the new?

Comment (MeBrei) [reply src/Radicle/Internal/ConcurrentMap.hs]: On the new... In the old code, `Map.lookup k m` in `lookup` returned `Nothing` when the key didn't exists, now, as it uses `modifyValue`, `Map.lookup k m` will return `pure (Map.insert k valueVar m, valueVar)`

Comment (jkarni) [reply src/Radicle/Internal/ConcurrentMap.hs]: Yeah, it does seem to me like @MeBrei is right and this leads to `Nothing`s being inserted.

Comment (jameshaydon) [reply src/Radicle/Internal/ConcurrentMap.hs]: Ah I see it adds useless `Nothing`s, agreed.

**Comment by jameshaydon** [comment]
I'm going to wait for the improved e2e before merging.

**Pull request merged**